### ISSUE TITLE
Hanging with "process.nextTick"

### ISSUE THREAD

**Issue by pilwon** [abbr/deasync#2]
`deasync` calls are hanging when calling a [Bluebird nodeified function: promise.nodiefy()](https://github.com/petkaantonov/bluebird/blob/master/API.md#nodeifyfunction-callback--object-options---promise) as seen in the following code sample:

```js
var Promise = require('bluebird');
var deasync = require('deasync');

function foo(cb) {
  return Promise.resolve()
    .then(function () {
      return 'bar';
    })
    .nodeify(cb);
}

// Prints 'bar'. No problem with `bluebird` library.
foo().then(console.log);

// Prints 'bar'. No problem with `bluebird`'s `promise.nodeify()`.
foo(function (err, result) {
  console.log(result);
});

// Hanging at `Promise.resolve()` when used with `deasync`...
console.log(deasync(foo)());
```

**Comment by pilwon** [comment]
Just discovered this was broken since `bluebird@2.9.3`. (works fine in v2.9.2)

**Comment by pilwon** [comment]
Narrowed down the problem to [this commit](https://github.com/petkaantonov/bluebird/commit/9fcbd99398dc5b8726fb4f1c6e20752e1b39e3e1).

I confirmed there's a bug in parsing `process.version` and I will open an issue at Bluebird project issues page. As a result it uses `process.nextTick` instead of `global.setImmediate`.

Question to @abbr: Is there a way to make `deasync` work properly even when `process.nextTick` is used?

**Comment by abbr** [comment]
I tried your code with Node v0.10.33, `bluebird@2.9.3` on [c9.io](https://c9.io), which runs on Ubuntu, and got expected 3 `bar` output without hung.
What environment are you on?

**Comment by pilwon** [comment]
@abbr It seems like it has something to do with Node.js engine version. The code works in v0.10 but fails in both v0.11 and v0.12.

**Comment by pilwon** [comment]
This is a simplified code independent from `bluebird`:

```js
var deasync = require('deasync');

function foo1(cb) {
  setImmediate(function () {
    cb(null, 'bar');
  });
}

function foo2(cb) {
  process.nextTick(function () {
    cb(null, 'bar');
  });
}

console.log(deasync(foo1)());  // works with `setImmediate`
console.log(deasync(foo2)());  // hanging with `process.nextTick`
```

**Comment by abbr** [comment]
I confirmed your observation and just published `deasync@0.0.9` to fix this problem. Will close the ticket for now. Let me know if you still got same problem in the new version.

**Comment by pilwon** [comment]
@abbr It works. Thanks for the fix-
